StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1642SA1643CodeFixProvider.cs

```diff
@@ -39,58 +40,66 @@ public override FixAllProvider GetFixAllProvider()
         /// <inheritdoc/>
         public override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
             foreach (var diagnostic in context.Diagnostics)
             {
                 if (!FixableDiagnostics.Contains(diagnostic.Id))
                     continue;
 
-                var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
                 var node = root.FindNode(diagnostic.Location.SourceSpan, findInsideTrivia: true) as XmlElementSyntax;
                 if (node == null)
                     continue;
-                var classDeclaration = node.FirstAncestorOrSelf<ClassDeclarationSyntax>();
-                var declarationSyntax = node.FirstAncestorOrSelf<BaseMethodDeclarationSyntax>();
 
-                ImmutableArray<string> standardText;
-                if (declarationSyntax is ConstructorDeclarationSyntax)
+                context.RegisterCodeFix(CodeAction.Create("Add standard text.", token => GetTransformedDocument(context, root, node)), diagnostic);
+            }
+        }
+
+        private static Task<Document> GetTransformedDocument(CodeFixContext context, SyntaxNode root, XmlElementSyntax node)
```

Comment:
:exclamation: Avoid passing the `CodeFixContext` instance to the callback. Its `CancellationToken` property and `RegisterCodeFix` method are not valid at this time and it would be too easy to mistakenly use them.